src/httpClient/httpClient.ts

```diff
@@ -177,6 +189,17 @@ export default class HttpClient {
     };
   }
 
+  // implemented based on https://github.com/RasCarlito/axios-cache-adapter/blob/master/src/cache.js#L77
+  public static generateCacheKey(req: AxiosRequestConfig): string {
+    const prefix = req.headers?.Authorization
+      ? parseJwtCanonicalId(req.headers.Authorization.replace('Bearer ', ''))
+      : 'shared';
```

Comment:
@mmaruniak this is where I'm defaulting to `shared` to use the same cache when no Authorization header was present. Do you think we should change this to UUID to be on the safe side with no-caching instead?

Reply:
I'm not sure, when there's no auth header the response will probably be the same for all callers. But I'd leave it up to you - I think both options are good

Reply:
Oh and what happens when there's no `'https://claims.cimpress.io/canonical_id'` in the token?

Reply:
I've updated the code, have a look and let me know if you see more flaws :)

Reply:
I like it!